compose-files/staff/docker-compose.yml

```diff
@@ -0,0 +1,44 @@
+version: "3.7"
+
+networks:
+  placeos_placeos:
```

Comment:
Thanks, so I tried this out and then did some troubleshooting with different configs (made sure to always delete the networks and containers prioir to each of `install` and have collected these results:

1. master branch ./install 
    - resulting networks: placeos   <- this is what we want


2. feat/staff-api branch ./install 
    - resulting networks: placeos


3. feat/staff-api branch ./install -a  (executes "docker-compose -f ./docker-compose.yaml -f staff/docker-compose.yaml up")
with docker-compose declaring
 networks:
  placeos:

    - resulting networks:
        placeos (empty)   <- init runs here so it fails
        placeos_placeos (has all containers)
> (it's as if docker-compose initially ignores the project name in .env for some reason) 

4. feat/staff-api branch ./install -a, 
with docker-compose declaring
networks:
  placeos_placeos:
    external: true

    - resulting networks:
        placeos (has all containers, so init succeeds)

here's the full output of scenario 3.:
```
=== Bringing up services...
WARNING: The SMTP_SERVER variable is not set. Defaulting to a blank string.
WARNING: The SMTP_USER variable is not set. Defaulting to a blank string.
WARNING: The SMTP_PASS variable is not set. Defaulting to a blank string.
WARNING: The SMTP_SECURE variable is not set. Defaulting to a blank string.
WARNING: The SMTP_SERVER variable is not set. Defaulting to a blank string.
WARNING: The SMTP_USER variable is not set. Defaulting to a blank string.
WARNING: The SMTP_PASS variable is not set. Defaulting to a blank string.
WARNING: The SMTP_SECURE variable is not set. Defaulting to a blank string.
WARNING: The Docker Engine you're using is running in swarm mode.

Creating network "placeos_placeos" with the default driver
Creating volume "placeos_elastic-data" with default driver
Creating volume "placeos_influx-data" with default driver
Creating volume "placeos_redis-data" with default driver
Creating volume "placeos_rethink-data" with default driver
Creating volume "placeos_www" with default driver
Creating volume "placeos_postgres-data" with default driver
Creating dispatch  ... done
Creating redis     ... done
Creating rethink  ... done
Creating postgres  ... done
Creating influx    ... done
Creating etcd      ... done
Creating elastic   ... done
Creating frontends ... done
Creating auth      ... done
Creating core      ... done
Creating triggers  ... done
Creating source    ... done
Creating staff     ... done
Creating rubber-soul ... done
Creating init        ... done
Creating api         ... done
Creating nginx       ... done
=== Configuring InfluxDB...
=== Initialising InfluxDB API
Config default has been stored in /root/.influxdbv2/configs.
User    Organization    Bucket
placeos PlaceOS         place
InfluxDB configured. API key is located in .env.influxdb.
org: PlaceOS
bucket: place
username: placeos
password: development
=== To setup sentry, run `./scripts/setup-sentry.sh`
=== To setup elastic stack run 'docker-compose up -d' in monitor-host directory.
=== Running init container
WARNING: The SMTP_SERVER variable is not set. Defaulting to a blank string.
WARNING: The SMTP_USER variable is not set. Defaulting to a blank string.
WARNING: The SMTP_PASS variable is not set. Defaulting to a blank string.
WARNING: The SMTP_SECURE variable is not set. Defaulting to a blank string.
Creating network "placeos" with driver "bridge"
level=ERROR time=2021-02-09T06:28:37Z source=place_os.tasks.entities message=Authority creation failed with: Hostname lookup for rethink failed: No address found (Socket::Addrinfo::Error)
```

It's weird the behaviour makes me second guess myself so I did a lot of testing.
I would expect
```
networks:
  placeos:
```
(without `external: true`) to work and I think that's what i tried originally when adding staff/docker-compose, but it ends up with the above results for some reason.

Do you guys get the same result?
This is with
```
w.le@w-aca-im ~/c/p/partner-environment (feat/staff-api)> docker version
Client: Docker Engine - Community
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 17:02:36 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:01:06 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

Reply:
So I had to check the behaviour on this too. At the moment this environment layers on the additional services for the staff API (and sentry, elk) via seperate compose files. When this happens [these get smooshed, with later files simply overriding any previously declared attributed](https://docs.docker.com/compose/extends/#understanding-multiple-compose-files).

This results in the network that you're defining in the staff api `docker-compose.yml` overriding what's in the base compose during the [service boot](https://github.com/place-labs/partner-environment/blob/df3924cea6fbf1def1115bec05c97f9919984206/install#L71). When the [init container runs](https://github.com/place-labs/partner-environment/blob/df3924cea6fbf1def1115bec05c97f9919984206/install#L96) this then only sees the original compose and original network conf.

Quick fix: drop the `networks` section from `staff/docker-compose.yml`. Alternatively, it may also be worth considering using [docker compose profiles](https://docs.docker.com/compose/profiles/).

Reply:
YES! let's use profiles!!!! couldn't agree with that more

Reply:
nice kim! thanks so much. i've now tested again and confirmed that the removal of the `networks` section from staff/docker-compose.yml allows both `install -a` and `install` to work fine